docs/HowToGuides/FirstPullRequest.md

```diff
@@ -0,0 +1,169 @@
+
+[create a pull request]: https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request#creating-the-pull-request
+
+## Asking for code review
+
+If you had an active discussion with someone on how to implement your change,
+you can `@` mention them in the PR description and ask for code review.
+If you directly implemented the change without any guidance from anyone else,
+`@` mention someone from GitHub's suggested reviewers. Please ask someone
+though! We don't want you to get stuck because you were not sure who to ask
+for code review.
+
+At the beginning, contributors are not able to run the continuous integration
+(CI) bot, which builds the project and runs tests. Please ask your code
+reviewer(s) to invoke the bot for you.
+
+## Responding to code review comments
+
+During the process of code review, someone might suggest changes or have
+questions about the implementation. If something is unclear, such as someone
+using a technical term you don't recognize, check our
+[Lexicon](/docs/Lexicon.md) or ask someone instead of trying to figure out
+everything by yourself. Code review does not need to be a one-way
+street. It is also a good opportunity for you to ask highly contextual
+questions on topics that you struggled with or were unable to understand.
+
+While making changes based on code review, if you are comfortable with
+rebasing, prefer rebasing and force-pushing for small patches (say < 100 lines).
+For larger patches, you can add fixup commits (`git commit --fixup ...`)
+addressing the suggestions and rebase after it the patch has been approved
+to clean up the history.
+
+When you push again and want the tests to be re-run, please ask the reviewer
+to invoke `swift-ci` for you.
+
+At the end, once the tests are passing, the pull request is approved by
+the reviewer, and you are satisfied with your changes, ask your reviewer
+to merge your changes. :tada:
+
+## I can't finish the contribution I started. :frowning_face:
+
+That's totally okay! There is no shame in that. You only have limited time and
+energy in a day. If you can, leave a comment on the bug report/pull request
+that you will not be able to continue and unassign yourself from the bug on
+JIRA. Don't worry about trying to explain _why_ you aren't able to contribute
+further. We understand. Unanticipated things come up all the time and you
+should do what _works for you_.
+
+This point also applies if you don't have time right now but hope to get to
+something in the near future. Please don't feel sad or apologetic!
+
+## I submitted and merged my first pull request. What now?
+
+Awesome! You could try fixing a few more Starter Bugs until you feel some
+level of comfort working with the codebase. You could also start looking at
+other bugs which interest you and you think you might be able to tackle.
+Don't forget to ask for help if you need directions or you get stuck!
```

Comment:
Worth discussing the process to be granted commit access, or is that overkill for a "first pull request" guide?

Reply:
There's a description of it on [swift.org](https://swift.org/contributing/#contributing-code), so I didn't think we should duplicate that here. Are you suggesting that I link it here?

Reply:
Yeah I wasn't thinking it needed full explication, a link to that process and brief mention would be fine, IMO. (Though, as I mentioned, I'm not sure concerns about commit access are necessarily appropriate for a "first pull request" guide at all.)

Reply:
I think it would be helpful to have a short explanation of commit access and the privileges/restrictions associated with it. For example, I was confused by not being able to summon the CI bot in my first pull request, because I did know it required commit access.

Reply:
This documentation does mention that:

1. You should @ mention people for code review (because you can't mark them as reviewer without being a collaborator).
2. You should @ mention people for CI (because you won't have CI privileges at the beginning).
3. You should @ mention people for merging (because you won't have write permissions).

I've tried to focus on what you should do, instead of what you can't do. Are you suggesting that I explicitly talk about what you can't do? Or that I consolidate those 3 pieces of information in one place somehow?

Reply:
Maybe it would be good to have a short discussion at the end to the effect of "once you have made sufficient contributions (blah blah blah) you can ask to be granted commit access, which will allow you to request reviewers, trigger the CI bot, and merge pull requests yourself." Right now the only indication that that can change is "at the beginning," which doesn't help someone who is wondering when they can have those restrictions lifted.

Reply:
@varungandhi-apple Sorry I misunderstood the discussion here. I thought you were going to remove all explanation of commit access, and only leave a link the swift.org. 

I think what you already wrote in the documentation is good enough to avoid confusion for first-time contributors. 

> Are you suggesting that I explicitly talk about what you can't do?

I didn't think of this at first, but now that you mentioned it, I think it might be good to include some "what you can't do" in addition to "what you should do", if there are things that you can't do and should't even try to do it by @ mentioning people.

Reply:
^^ Sorry, yeah, I got that part. I will include that, need to think a little bit about phrasing. I was asking for clarification on Wowbaggers's question.

> I didn't think of this at first, but now that you mentioned it, I think it might be good to include some "what you can't do" in addition to "what you should do", if there are things that you can't do and should't even try to do it by @ mentioning people.

I do have a sentence for the CI, since that's the thing that's not clear from the UI, and something I've personally seen people get repeatedly confused over.

> At the beginning, contributors are not able to run the continuous integration (CI) bot, which builds the project and runs tests. Please ask your code reviewer(s) to invoke the bot for you.

You might mimic someone invoking the bot, but nothing happens, there is no feedback. OTOH, with code review, you will not see the option to select reviewers, and with merging, the button will be greyed out/not show up. So there is already feedback in the UI which explains things. I don't think repeating that information in text is worth it.

We can certainly add it in later, if we see contributors getting confused over those two things.